async_geoprocess/src/cumulus_geoproc/geoprocess/snodas/interpolate.py

```diff
@@ -198,6 +199,20 @@ async def snodas(cfg: namedtuple, dst: str):
 
         max_dist = 0 if code in no_interp else cfg.max_distance
 
+        # check to see if the file exists
```

Comment:
Suggest not using the approach of trying a S3 HEAD request. The file prefix of `zz_` or `us_` is known (or can be determined), based on the `dt`.  We only download/process the masked dataset (prefix `us_`) when the unmasked dataset (prefix `zz_`) was not available yet - prior to ~2010.  Recommend modifying the strings.Template for each product to allow us_ or zz_, depending on the date.

Reply:
I'll take a look at it.

Reply:
Awesome, thanks @jeffsuperglide